### PR TITLE
chore: single validator DHIS2-14298

### PR DESCRIPTION
**Pull request by teleivo**
* first commit: create one main `Validator<TrackerBundle>` out of the per type specific ones like `Validator<Enrollment>`
* second commit: replace hook(s) with validator as the old hook interface does not exist anymore

### CODE REVIEW

File: dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/validation/DefaultValidationService.java [hunk 1]
```diff
@@ -49,24 +49,24 @@
 {
 
     @Qualifier( "org.hisp.dhis.tracker.validation.DefaultValidators" )
-    private final Validators validators;
+    private final Validator<TrackerBundle> validator;
```
Comment (enricocolasante) [inline dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/validation/DefaultValidationService.java:52]: I think the qualifiers are wrong.
org.hisp.dhis.tracker.validation.validator.DefaultValidator should be on validator
and org.hisp.dhis.tracker.validation.validator.RuleEngineValidator should be on ruleEngineValidator

**Review by enricocolasante — COMMENTED** [review]
Check the qualifiers for validator components

File: dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/validation/DefaultValidationService.java (hunk 1 above)
Comment (teleivo) [reply dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/validation/DefaultValidationService.java:52]: Thank you for spotting this 😂

**Review by enricocolasante — APPROVED** [review]
Check the qualifiers for validator components

**Review by stian-sandvold — APPROVED** [review]

**Review by stian-sandvold — APPROVED** [review]

**Pull request merged**